packages/backend/src/models/ProjectModel/ProjectModel.ts

```diff
@@ -752,41 +750,42 @@ export class ProjectModel {
                                 cachedExplores,
                             ),
                     );
-                    await wrapSentryTransaction(
+
+                    const transactionInserts = await wrapSentryTransaction(
                         'indexCatalog.insert',
                         { projectUuid, catalogSize: catalogItems.length },
-                        async () =>
+                        () =>
                             this.database.transaction(async (trx) => {
                                 await trx(CatalogTableName)
                                     .where('project_uuid', projectUuid)
                                     .delete();
 
-                                // Insert in chunks to avoid having a massive insert query
-                                const chunkSize = 1000;
-                                const chunkedInsertPromises = [];
-                                for (
-                                    let i = 0;
-                                    i < catalogItems.length;
-                                    i += chunkSize
-                                ) {
-                                    const chunk = catalogItems
-                                        .slice(i, i + chunkSize)
-                                        .map((catalogItem) => ({
+                                const inserts = await this.database
+                                    .batchInsert(
```

Comment:
nice, I didn't know about this utility function. 

> By default, the chunkSize is set to 1000.

So in theory it should be the same.

Reply:
@rephus I'll DM the details